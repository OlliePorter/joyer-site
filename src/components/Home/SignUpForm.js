import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import theme from '../../theme';

const useStyles = makeStyles({
  signUpBackground: {
    height: '100%',
    backgroundColor: theme.palette.secondary.light,
    textAlign: 'center',
    padding: '70px',
  },
  title: {
    color: theme.palette.secondary.dark,
    fontWeight: 300,
    padding: '10px',
  },
  subtitle: {
    color: theme.palette.secondary.dark,
    padding: '8px',
    fontWeight: 300,
    paddingBottom: '70px',
  },
  nameInput: {
    padding: '8px 15px',
    lineHeight: '1.428571429',
    margin: '15px',
    border: '1px solid #cccc',
    borderRadius: '2px',
    textShadow: 'none',
    height: '20px',
    width: '150px',
    fontSize: '15px',
    fontFamily: 'roboto',
    "&:focus": {
      outline: 'none',
      border: '1px solid #49d0be',
    },
  },
  emailInput: {
    padding: '8px 16px',
    lineHeight: '1.428571429',
    margin: '15px',
    border: '1px solid #cccc',
    borderRadius: '2px',
    textShadow: 'none',
    height: '20px',
    width: '360px',
    fontSize: '15px',
    fontFamily: 'roboto',
    "&:focus": {
      outline: 'none',
      border: '1px solid #49d0be',
    },
  },
  signUpButton: {
    fontSize: '15px',
    borderRadius: '24px',
    width: '150px',
    height: '40px',
    padding: '8px 16px',
    fontFamily: 'roboto',
    fontWeight: 400,
    borderColor: '#1e1e1e',
    backgroundColor: '#444',
    color: 'white',
    margin: '15px',
    cursor: 'pointer',
    verticalAlign: 'middle',
    "&:hover": {
      backgroundColor: '#333',
    },
  }
});

export default function SignUpForm() {
  const classes = useStyles();

    return (
    <div className={classes.signUpBackground}>
      <Typography component="h2" variant="h4" className={classes.title}>
      Become a Joyager
      </Typography>
      <Typography component="p" className={classes.subtitle}>
      Subscribe for the latest updates.
      </Typography>
      <form action="https://joyerapp.us20.list-manage.com/subscribe/post?u=9c4cca26d3bea1021320b9967&id=10b0082a7c" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" target="_blank">
        <div className="mc-field-group">
          <input type="text" name="FNAME" className={classes.nameInput} id="mce-FNAME" placeholder="First Name"/>
          <input type="text" name="LNAME" className={classes.nameInput} id="mce-LNAME" placeholder="Last Name"/>
        </div>
        <div>
          <input type="email" name="EMAIL" className={classes.emailInput} id="mce-EMAIL" placeholder="Email"/>
        </div>
        <div>
          <input type="submit" name="subscribe" id="mc-embedded-subscribe" value="Sign Up" className={classes.signUpButton}/>
        </div>
      </form>
    </div>
    );
}
