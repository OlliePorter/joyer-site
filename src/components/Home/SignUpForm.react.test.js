import React from 'react';
import SignUpForm from './SignUpForm';
import renderer from 'react-test-renderer';

it('SignUpForm renders correctly', () => {
    const tree = renderer.create(
      <SignUpForm />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });