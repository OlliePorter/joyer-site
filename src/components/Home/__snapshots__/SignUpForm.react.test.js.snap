// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders correctly 1`] = `
<div
  className="makeStyles-signUpBackground-1"
>
  <h2
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h4"
  >
    Become a Joyager
  </h2>
  <p
    className="MuiTypography-root makeStyles-subtitle-3 MuiTypography-body1"
  >
    Subscribe for the latest updates.
  </p>
  <form
    action="https://joyerapp.us20.list-manage.com/subscribe/post?u=9c4cca26d3bea1021320b9967&id=10b0082a7c"
    id="mc-embedded-subscribe-form"
    method="post"
    name="mc-embedded-subscribe-form"
    target="_blank"
  >
    <div
      className="mc-field-group"
    >
      <input
        className="makeStyles-nameInput-4"
        id="mce-FNAME"
        name="FNAME"
        placeholder="First Name"
        type="text"
      />
      <input
        className="makeStyles-nameInput-4"
        id="mce-LNAME"
        name="LNAME"
        placeholder="Last Name"
        type="text"
      />
    </div>
    <div>
      <input
        className="makeStyles-emailInput-5"
        id="mce-EMAIL"
        name="EMAIL"
        placeholder="Email"
        type="email"
      />
    </div>
    <div>
      <input
        className="makeStyles-signUpButton-6"
        id="mc-embedded-subscribe"
        name="subscribe"
        type="submit"
        value="Sign Up"
      />
    </div>
  </form>
</div>
`;
